CONTRIBUTING.md

```diff
@@ -97,46 +97,33 @@ git fetch --all
 
 # Create the new release branch from the one on crossplane/crossplane.
 git checkout -b $RELEASE_BRANCH upstream/$RELEASE_BRANCH
+git reset --hard vX.Y.0 # or whatever version we are releasing
```

Comment:
fair point, thanks @chrishalbert
```suggestion
git diff --exit-code && git reset --hard vX.Y.0 # or whatever version we are releasing
```
@turkenh I would also add a variable above like `RELEASE_TAG` or something like that, wdyt?